scripts/docs/json_builder.py

```diff
@@ -0,0 +1,345 @@
+        obj_module = inspect.getmodule(obj)
+        if obj_module is None:
+            continue
+
+        if isinstance(obj, type):
+            # Filter out non-gooddata libraries
+            if MODULE_NAME in obj_module.__name__:
+                data[name] = class_data(obj)
+        elif isinstance(obj, ModuleType):
+            if MODULE_NAME in obj_module.__name__:
+                data[name] = module_data(obj)
+    return data
+
+
+def parse_package(obj: ModuleType, data: dict | None = None) -> dict:
+    """
+    Parse the package and its submodules into a dict object, that
+    can be converted into a json
+
+    Args:
+        obj (ModuleType): package object
+        data (dict): optional parameter for recursive calling
+    Returns:
+        dict: data of package
+
+
+    Example:
+        {
+        "submodule1": {
+            "file1": {
+                "class_name" : class_data (see `object_data`)
+                }
+            }
+        }
+    """
+    if not data:
+        data = {}
+    data["kind"] = "module"
+    if isinstance(obj, ModuleType):
+        iterator = pkgutil.iter_modules(obj.__path__)
+        for item in iterator:
+            if item.name not in data:
+                data[item.name] = {}
+
+            if item.ispkg:
+                data[item.name].update(parse_package(vars(obj)[item.name]))
+            else:
+                module = vars(obj)[item.name]
+                data[item.name].update(module_data(module))
+
+    return data
+
+
+def import_submodules(pkg_name: str) -> dict[str, ModuleType]:
+    """
+    Import all submodules of a package, enabling their parsing
+
+    Args:
+        pkg_name (str): package name
+    """
+    package = sys.modules[pkg_name]
+
+    return {
+        name: importlib.import_module(f"{pkg_name}.{name}")
+        for loader, name, is_pkg in pkgutil.walk_packages(package.__path__)
+    }
+
+
+def generate_links(module_data: dict) -> Dict[str, dict[str, str]]:
+    """
+    Generate links for the objects for the json data:
+
+    Args:
+        module_data(dict): dict with the module data
+    Returns:
+        dict: dict with the links
+    Example:
+    {
+        "AFM" : {"path": "compute.model.attribute.afm_models.AFM", "kind": "class"},
+        "Class2" : {"path": "path.in.json.Class2", "kind": "class"},
+        "func1" : {"path": "path.in.json.func1", "kind": "function"}
+    }
+    """
+    links = {}
+
+    def _recursive_find_classes(data: dict, path: str):
+        if data.get("kind", None) in ["class", "function"]:
+            # Last item in path corresponds to the class name
+            links[path.split(".")[-1]] = {"path": path, "kind": data["kind"]}
+
+        for key, val in data.items():
+            if isinstance(val, dict):
+                if key == "functions":
+                    _recursive_find_classes(val, path)
+                elif path != "":
+                    _recursive_find_classes(val, f"{path}.{key}")
```

Comment:
I do not fully understand this elif part? If `path != ""` then this branch is hit always, right? `

Can we do something like `path_key = f"{path}.{key}"`?